gorb/gorb_test.go

```diff
@@ -0,0 +1,149 @@
+package gorb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sky-uk/feed/controller"
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2E Suite")
+}
+
+const (
+	instanceIP       = "10.10.0.1"
+	drainImmediately = 0
+	servicesName     = "http-proxy"
+	servicesPort     = "80"
+	backendWeight    = 1000
+	backendMethod    = "dr"
+	vipLoadbalancer  = "127.0.0.1"
+)
+
+type gorbResponsePrimer struct {
+	response   string
+	statusCode int
+}
+
+type gorbRecordedRequest struct {
+	url  *url.URL
+	body *Backend
+}
+
+type gorbHandler struct {
+	responsePrimer  []gorbResponsePrimer
+	recordedRequest []gorbRecordedRequest
+	requestCounter  int
+}
```

Comment:
I think you need to make all accesses to this struct thread safe, or arrange the test code so it's thread safe. Otherwise you might get flaky failures.

Reply:
A way to do that for example would be to set the responsePrimer field before starting up the http server.